components/lib/datatable/DataTable.js

```diff
@@ -514,6 +514,9 @@ export const DataTable = React.forwardRef((props, ref) => {
                     !!el && (el.style.width = el.style.minWidth = tableWidth);
                 };
 
+                // https://github.com/primefaces/primereact/issues/3970
+                // Reasoning: resize table cells before updating the table width so that it can use existing computed cell widths and adjust only the one column. 
```

Comment:
Change this comment to just a single line `// #3970: resize table cells before updating the table width so that it can use existing computed cell widths and adjust only the one column.`

Reply:
Done.